app/src/main/java/agency/digitera/android/promdate/ui/SinglesTabFragment.kt

```diff
@@ -74,16 +77,18 @@ class SinglesTabFragment : Fragment(), TabInterface {
 
         liveData.observe(this, Observer<PagedList<User>> { pagedList ->
             viewAdapter.submitList(pagedList)
+            firstVisiblePos?.let {
+                recyclerView.layoutManager?.scrollToPosition(it)
+                firstVisiblePos = null
+            }
         })
     }
 
-    override fun onSaveInstanceState(outState: Bundle) {
-        super.onSaveInstanceState(outState)
-
-        val lastKey = viewAdapter.currentList?.lastKey as Int
-
-        outState.putInt("lastKey", lastKey)
-        outState.putParcelable("layout_manager_state", recyclerView.layoutManager?.onSaveInstanceState())
+    override fun onPause() {
+        super.onPause()
+        if ((recyclerView.layoutManager as LinearLayoutManager).findFirstCompletelyVisibleItemPosition() > 0)
```

Comment:
Just a basic suggestion =) 

```kotlin
val firstItemVisiblePosition: Int = (recyclerView.layoutManager as LinearLayoutManager).findFirstCompletelyVisibleItemPosition()
if (firstItemVisiblePosition > 0)
  firstVisiblePos = firstItemVisiblePosition
```

Reply:
You can also omit the type of `firstItemVisiblePosition`. Kotlin is able to infer the type. This will help you with a cleaner code.

Reply:
Done!